protobuf/lib/src/protobuf/proto3_json.dart

```diff
@@ -400,3 +406,48 @@ void _mergeFromProto3Json(
 
   recursionHelper(json, fieldSet);
 }
+
+bool _isAsciiLetter(int char) {
+  const lowerA = 97;
+  const lowerZ = 122;
+  const capitalA = 65;
+  const capitalZ = 90;
+
+  return (capitalA <= char && char <= capitalZ) ||
+      (lowerA <= char && char <= lowerZ);
+}
+
+/// Returns true if [a] and [b] are the same ignoring case and all instances of
+///  `-` and `_`.
+bool _permissiveCompare(String a, String b) {
+  const dash = 45;
+  const underscore = 95;
+
+  // Enum names are always ascii.
+  int i = 0;
+  int j = 0;
+
+  while (i < a.length && j < b.length) {
+    int ca = a.codeUnitAt(i);
+    if (ca == dash || ca == underscore) {
+      i++; continue;
+    }
+    int cb = b.codeUnitAt(j);
+    if (cb == dash || cb == underscore) {
+      j++; continue;
+    }
+
+    if (_isAsciiLetter(ca)) {
+      if (ca | 0x20 != cb | 0x20) {
+        return false;
+      }
+    } else {
+      if (ca != cb) {
+        return false;
+      }
+    }
```

Comment:
How about (because I loooove optimizing):
```dart
if (ca != cb && (ca ^ cb != 0x20 || !_isAsciiLetter(ca))) return false;
```

Reply:
I'll buy it!